flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartitionWithReadViewTest.java

```diff
@@ -353,14 +354,32 @@ private void testBacklogConsistentWithNumberOfConsumableBuffers(boolean isFlushR
 		}
 	}
 
+	@Test
+	public void testBlockedByCheckpointAndResumeConsumption() throws IOException, InterruptedException {
+		subpartition.add(createFilledFinishedBufferConsumer(BUFFER_SIZE));
+		subpartition.add(createEventBufferConsumer(BUFFER_SIZE, Buffer.DataType.EXACTLY_ONCE_CHECKPOINT_BARRIER));
+		subpartition.add(createEventBufferConsumer(BUFFER_SIZE, Buffer.DataType.EVENT_BUFFER));
+		subpartition.add(createFilledFinishedBufferConsumer(BUFFER_SIZE));
+		subpartition.finish();
+
+		assertNextBuffer(readView, BUFFER_SIZE, true, 1, true, true);
+		assertNextEvent(readView, BUFFER_SIZE, null, false, 1, false, true);
```

Comment:
why we can not verify the real event class here and pass `null` for all the events, otherwise it is no need to introduce this `expectedEventClass` argument in `assertNextEvent`

Reply:
The expectedEventClass is never used currently, and if we want to check the event class, we need to create a real event, however, we use fake events only.